radio/src/gui/colorlcd/model_setup.cpp

```diff
@@ -1337,8 +1337,83 @@ void ModelSetupPage::build(FormWindow * window)
       }
     }
     grid.addWindow(group);
+
+    // Pots and sliders warning
+    if (NUM_POTS + NUM_SLIDERS) {
+      new StaticText(window, grid.getLabelSlot(true), STR_POTWARNINGSTATE);
+      new Choice(window, grid.getFieldSlot(), {"OFF", "ON", "AUTO"}, 0, 2,
+                 GET_SET_DEFAULT(g_model.potsWarnMode));
+      grid.nextLine();
+
+      if (NUM_POTS) {
+        new StaticText(window, grid.getLabelSlot(true), STR_POTWARNING);
+        auto group =
+            new FormGroup(window, grid.getFieldSlot(),
+                          FORM_BORDER_FOCUS_ONLY | PAINT_CHILDREN_FIRST);
+        GridLayout centerGrid(group);
+        for (int i = POT_FIRST, j = 0; i <= POT_LAST; i++) {
+          char s[8];
+          if ((IS_POT(i) || IS_POT_MULTIPOS(i))) {
+            if (j > 0 && ((j % 4) == 0)) centerGrid.nextLine();
+
+            auto button = new TextButton(
+                group, centerGrid.getSlot(4, j % 4),
+                getStringAtIndex(s, STR_VSRCRAW, i + 1), nullptr,
+                OPAQUE | ((g_model.potsWarnEnabled & (1 << j)) ? BUTTON_CHECKED
+                                                               : 0));
+            button->setPressHandler([button, j] {
+              g_model.potsWarnEnabled ^= (1 << j);
+              if ((g_model.potsWarnMode == POTS_WARN_MANUAL) &&
+                  (g_model.potsWarnEnabled & (1 << j))) {
+                SAVE_POT_POSITION(j);
+              }
+              button->check(g_model.potsWarnEnabled & (1 << j));
+              SET_DIRTY();
+              return (g_model.potsWarnEnabled & (1 << j));
+            });
+            j++;
+          }
+        }
+        grid.addWindow(group);
+      }
+
+      if (NUM_SLIDERS) {
```

Comment:
I haven't confirmed this but since NUM_SLIDERS is #defined and not a variable the compiler might be taking a path not expected. if (0) is not the same as #if 0. Try NUM_SLIDERS > 0 and if this works change the similar tests above. Hope my guess works.

Reply:
Spot on... changing to `NUM_SLIDERS > 0` lets this continue, and splatter on needing the `SLIDER_FIRST` & `SLIDER_LAST` to exist (I just initalised to zero since they don't exist) and it then compiled to completion.